File: src/main/java/ch/obermuhlner/astro/gradient/operation/SubtractImageOperation.java
package ch.obermuhlner.astro.gradient.operation;

public class SubtractImageOperation extends AbstractSimpleChannelImageOperation {

  @Override
  protected double channelOperation(double channel1, double channel2, int x, int y, int channelIndex) {
    return channel1 - channel2;
  }

  @Override
  public String toString() {
    return "Subtract";
  }
}

File: src/main/java/ch/obermuhlner/astro/gradient/operation/SubtractImageOperation.kt
package ch.obermuhlner.astro.gradient.operation

class SubtractImageOperation : AbstractSimpleChannelImageOperation() {
    override fun channelOperation(channel1: Double, channel2: Double, x: Int, y: Int, channelIndex: Int): Double {
        return channel1 - channel2
    }

    override fun toString(): String {
        return "Subtract"
    }
}